### PR TITLE
Make all amm and integration pass on windows except the Syntax Error ones

### PR DESCRIPTION
**Pull request by coderabhishek**
Major Changes:
1) To make the extended ascii character things defined in `scalaz` and similar other problems with `shapeless` we need to set encoding to `utf-8` on windows, for which I have set environment var on appveyor     `JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8`

2) There are no other major changes except one liners like 

```
private def jarSignature(url: URL) = {
-    val path = Path(url.getFile, root)
+    val path = Path(file.Paths.get(url.toURI()).toFile(), root)
```

3) Few ivy tests are behaving unusual on appveyor but thats just caching problem so I have made `appveyor.yml` itself a dependency for `.ivy2` cache so any major change in appveyor will redownload ivy. Though I am not sure if this change should merge into master.
4) I have introduced a `sanitizePath` function in `Util.scala` to remove windows or linux restricted chars from file names and replace them by `$foo`.
5) Rest is just uncommenting `if(!Util.windowsPlatform)` guard etc.

### CODE REVIEW

File: amm/src/main/scala/ammonite/interp/ClassLoaders.scala [hunk 1]
```diff
@@ -118,7 +118,7 @@ class SpecialClassLoader(parent: ClassLoader, parentSignature: Seq[(Path, Long)]
   }
 
   private def jarSignature(url: URL) = {
-    val path = Path(url.getFile, root)
```
Comment (lihaoyi) [inline amm/src/main/scala/ammonite/interp/ClassLoaders.scala]: What is the purpose of this change? You obviously thought it was interesting enough to mention in the description but you didn't actually say why it was necessary

File: amm/src/main/scala/ammonite/util/Util.scala [hunk 2]
```diff
@@ -58,6 +58,19 @@ object Util{
   //irrespective of the OS on which script was written
   def normalizeNewlines(s: String) = s.replace("\r", "").replace("\n", newLine)
 
+  def sanitizePath(s: String) =
```
Comment (lihaoyi) [inline amm/src/main/scala/ammonite/util/Util.scala]: Can we implement this using `NameTransformer.encode` or `decode`? We use those in a few places like https://github.com/lihaoyi/Ammonite/blob/9e249f6d5fd97c56121f11ca77b59071e4a72e25/amm/src/main/scala/ammonite/util/Model.scala#L117, and it's the way the Scala compiler currently does it, so if possible it would nice to be consistent with them

File: integration/src/test/scala/ammonite/integration/ErrorTruncationTests.scala [hunk 3]
```diff
@@ -18,7 +18,7 @@ object ErrorTruncationTests extends TestSuite{
     val e = fansi.Str(
       intercept[ShelloutException]{ exec(file) }.result.err.string
     ).plainText
-    assert(e == expected)
```
Comment (lihaoyi) [inline integration/src/test/scala/ammonite/integration/ErrorTruncationTests.scala]: Why is this change necessary? The goal of this test suite is to enforce that these outputs are short and do not contain unnecessary stack traces, and this change would invalidate that. If you want to make it less strict on windows, gate your change with a `Util.windowsPlatform` to allow it to pass less-strictly on windows while still being strict on linux

File: appveyor.yml [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # Adapted from https://github.com/sbt/sbt-native-packager/blob/master/appveyor.yml
 version: '{build}'
 os: Windows Server 2012
+environment:
+        JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
```
Comment (lihaoyi) [inline appveyor.yml]: This deserves a comment as to why it's necessary

**Comment by lihaoyi** [comment]
That is all for now!

File: integration/src/test/scala/ammonite/integration/ErrorTruncationTests.scala (hunk 3 above)
Comment (lihaoyi) [inline integration/src/test/scala/ammonite/integration/ErrorTruncationTests.scala]: Also, when you do make this change, leave a comment explaining why it is necessary. This sort of special-casing is entirely non-obvious and when you come back a year later it's easy to not know why it was done

File: amm/src/main/scala/ammonite/interp/ClassLoaders.scala (hunk 1 above)
Comment (coderabhishek) [inline amm/src/main/scala/ammonite/interp/ClassLoaders.scala]: Actually `url.getFile` doesn't converts urls back to OS specific path.
http://stackoverflow.com/questions/6164448/convert-url-to-normal-windows-filename-java

Also `url.getFile` is now deprecated

File: integration/src/test/scala/ammonite/integration/ErrorTruncationTests.scala (hunk 3 above)
Comment (coderabhishek) [inline integration/src/test/scala/ammonite/integration/ErrorTruncationTests.scala]: I think the new change should be fine. Is it good enough?

Comment (lihaoyi) [inline integration/src/test/scala/ammonite/integration/ErrorTruncationTests.scala]: ok

File: amm/src/main/scala/ammonite/interp/ClassLoaders.scala (hunk 1 above)
Comment (lihaoyi) [inline amm/src/main/scala/ammonite/interp/ClassLoaders.scala]: ok

**Comment by lihaoyi** [comment]
Looks good to me!
